upup/pkg/fi/cloudup/awstasks/dnszoneassociation.go

```diff
@@ -0,0 +1,119 @@
+package awstasks
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/golang/glog"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
+	"k8s.io/kops/upup/pkg/fi/cloudup/terraform"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+)
+
+//go:generate fitask -type=DNSZoneAssociation
+type DNSZoneAssociation struct {
+	Name    *string
+	DNSZone *DNSZone
+	VPC     *VPC
+}
+
+var _ fi.CompareWithID = &AutoscalingGroup{}
+
+func (e *DNSZoneAssociation) CompareWithID() *string {
+	return e.Name
+}
+
+func (e *DNSZoneAssociation) Find(c *fi.Context) (*DNSZoneAssociation, error) {
```

Comment:
We either return `found, nil` or `nil, error` or `nil, nil`.  The last case is not found, no error (i.e. does not exist).  I think that covers the 3 cases pretty nicely.